python/pyarrow/_flight.pyx

```diff
@@ -874,6 +874,13 @@ cdef class _MetadataRecordBatchReader(_Weakrefable, _ReadPandasMixin):
 
         return chunk
 
+    def to_reader(self):
+        """Convert this reader into a regular RecordBatchReader."""
```

Comment:
Nit: Perhaps add a note that it does this by first reading the schema or some kind of indication that calling `to_reader` will cause it to read the schema.